File: frontend/src/types/agent.ts
export interface AgentConfig {
  id: string;
  name: string;
  display_name: string;
  icon: string;
  description: string;
  available: boolean;
  skills: string[];
  preferred_agent: string;
  agent_type: "copilot" | "system";
  modes?: string[];
  version?: string;
  provider?: string;
  model_name?: string;
  install_hint?: string;
}

export interface AgentSession {
  id: string;
  agent_name: string;
  title: string;
  created_at: string;
}

export interface AgentMessage {
  id: string;
  role: "user" | "assistant" | "system";
  content: string;
  created_at: string;
}

export interface AgentEvent {
  type: "text" | "error" | "done" | "session";
  content?: string;
  session_id?: string;
}

export interface AgentConfigAdmin {
  id: string;
  name: string;
  display_name: string;
  icon: string;
  description: string;
  system_prompt: string;
  skills: string[];
  preferred_agent: string;
  fallback_agents: string[];
  available: boolean;
  model_config_json: Record<string, unknown>;
  agent_type: "copilot" | "system";
  modes?: string[];
  version?: string;
  provider?: string;
  model_name?: string;
  install_hint?: string;
  tools?: string[];
  mcp_servers?: string[];
  created_at: string;
  updated_at: string;
}

export interface AgentConfigCreate {
  name: string;
  display_name: string;
  icon?: string;
  description?: string;
  system_prompt?: string;
  skills?: string[];
  preferred_agent?: string;
  fallback_agents?: string[];
  available?: boolean;
  model_config_json?: Record<string, unknown>;
  agent_type?: "copilot" | "system";
  modes?: string[];
  version?: string;
  provider?: string;
  model_name?: string;
  install_hint?: string;
  tools?: string[];
  mcp_servers?: string[];
}

export interface AgentConfigUpdate {
  display_name?: string;
  icon?: string;
  description?: string;
  system_prompt?: string;
  skills?: string[];
  preferred_agent?: string;
  fallback_agents?: string[];
  available?: boolean;
  model_config_json?: Record<string, unknown>;
  agent_type?: "copilot" | "system";
  modes?: string[];
  version?: string;
  provider?: string;
  model_name?: string;
  install_hint?: string;
  tools?: string[];
  mcp_servers?: string[];
}

export interface AdapterInfo {
  name: string;
  available: boolean;
}

File: backend/app/schemas/agent.py
from datetime import datetime
from pydantic import BaseModel


class AgentConfigResponse(BaseModel):
    id: str
    name: str
    display_name: str
    icon: str
    description: str
    available: bool
    skills: list[str]
    preferred_agent: str
    agent_type: str = "copilot"
    modes: list[str] | None = None
    version: str | None = None
    provider: str | None = None
    model_name: str | None = None
    install_hint: str | None = None

    model_config = {"from_attributes": True, "protected_namespaces": ()}


class AgentConfigDetail(AgentConfigResponse):
    system_prompt: str
    fallback_agents: list[str]
    model_config_json: dict


class CreateSessionRequest(BaseModel):
    agent_name: str


class SessionResponse(BaseModel):
    id: str
    agent_name: str
    title: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ChatRequest(BaseModel):
    message: str
    session_id: str | None = None


class MessageResponse(BaseModel):
    id: str
    role: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AgentEventResponse(BaseModel):
    type: str  # text | error | done
    content: str = ""
    session_id: str = ""
